backend/routes/reservation.js

```diff
@@ -0,0 +1,101 @@
+import express from 'express';
+import bodyParser from 'body-parser';
+
+import connection from '../database';
+
+const router = express.Router();
+router.use(bodyParser.urlencoded({ extended: true }));
+
+// Retrieve a single reservation for a user to check their reservation.
+router.get('/single', (req, res) => {
+  const { reservationID } = req.query;
+
+  if (!reservationID) {
+    res.status(400).json({ error: 'reservation/single GET endpoint needs a reservationID query param' });
+    return;
+  }
+
+  connection.query(
+    `SELECT ` +
+      `ID, Date, Time, Notes, NumberOfGuests, TableID, RestaurantID, UserID FROM RESERVATION ` +
+      `WHERE ID = ? `,
+    [reservationID],
+    (error, results) => {
+      if (error) {
+        res.status(400).json({ error });
+        return;
+      }
+      res.json({ result: results });
+    }
+  );
+});
+
+// Retrieve all reservation from a single restaurant. Used by restaurant manager to see all reservations.
+router.get('/all', (req, res) => {
+  const { restaurantID } = req.query;
+
+  if (!restaurantID) {
+    res.status(400).json({ error: 'reservation/all GET endpoint needs a restaurantID query param' });
+    return;
+  }
+
+  connection.query(
+    `SELECT ID, Date, Time, Notes, NumberOfGuests, TableID, RestaurantID, UserID FROM RESERVATION WHERE RestaurantID = ?;`,
+    [restaurantID],
+    (error, results) => {
+      if (error) {
+        res.status(400).json({ error });
+        return;
+      }
+      res.json({ result: results });
+    }
+  );
+});
+
+// Add a new reservation when a user wants to book a table.
+router.post('/single', (req, res) => {
+  // Generate a random number capped at 2147483647 as this is the largest number the database can hold.
```

Comment:
What happens if we get an ID conflict?

Reply:
Currently, it will just return an error complaining about duplicate keys. To remove this possibility, we could change the ID to be varchar and use a package to generate non-duplicate ids.

Reply:
Do you think an issue would be good for this? How much would need to change if bookingID was changed to varchar? If it seems feasible, could you make an issue later and pin `needs-review` tag to it please?